### DISCUSSION TITLE
problem with X axis and cloned X axis.

### DISCUSSION THREAD

**Discussion by attk3** [fra589/grbl-Mega-5X#379] [General]
Hello good day. I am using mega 2560, ramps 1.4, 5 drv8825, along with grbl mega 5x. I have a mpcnc that uses 2 motors on the X and Y axes, and I cloned the X axis in A and the Y axis in B.
The cnc works but both the cloned A axis and B axis turn and rotate. I don't know how to solve this problem, if someone can help me it would be great since I'm stuck for several days without being able to solve it. Thanks!.
I leave my config.h file

// To use with RAMPS 1.4 Board, comment out the above defines and uncomment the next two defines
#define DEFAULTS_RAMPS_BOARD
#define CPU_MAP_2560_RAMPS_BOARD

// Serial baud rate
// #define BAUD_RATE 230400
#define BAUD_RATE 115200

// Axis array index values. Must start with 0 and be continuous.
#ifdef DEFAULTS_RAMPS_BOARD
  // 4, 5 & 6 axis support only for RAMPS 1.4 (for the moment :-)...)
  #define N_AXIS 5            // Number of axes
  #define N_AXIS_LINEAR 5     // Number of linears axis
#else
  #define N_AXIS 3 // Number of axes = 3 if not DEFAULTS_RAMPS_BOARD
#endif

#define AXIS_1 0        // Axis indexing value. Must start with 0 and be continuous.
#define AXIS_1_NAME 'X' // Axis names must be in X, Y, Z, A, B, C, U, V & W.
#define AXIS_2 1
#define AXIS_2_NAME 'Y'
#define AXIS_3 2
#define AXIS_3_NAME 'Z'

#if N_AXIS <3
  #error "N_AXIS must be >= 3. N_AXIS < 3 is not implemented."
#endif
#if N_AXIS > 3
  #define AXIS_4 3
  #define AXIS_4_NAME 'X' // Letter of axis number 4
#endif
#if N_AXIS > 4
  #define AXIS_5 4
  #define AXIS_5_NAME 'Y' // Letter of axis number 5
#endif
#if N_AXIS > 5
  #define AXIS_6 5
  #define AXIS_6_NAME 'C' // Letter of axis number 6
#endif
#if N_AXIS > 6
  #error "N_AXIS must be <= 6. N_AXIS > 6 is not implemented."
#endif

**Comment by attk3** [comment]
I leave the result of the Grbl commands $$ , $I

>>> $I
[VER:1.1l.20190605:]
[AXS:5:XYZXY]
[OPT:VNM,35,255,48]
ok

>>> $$
$0 = 10    (Step pulse time, microseconds)
$1 = 255    (Step idle delay, milliseconds)
$2 = 0    (Step pulse invert, mask)
$3 = 28    (Step direction invert, mask)
$4 = 0    (Invert step enable pin, boolean)
$5 = 0    (Invert limit pins, boolean)
$6 = 0    (Invert probe pin, boolean)
$10 = 1    (Status report options, mask)
$11 = 0.020    (Junction deviation, millimeters)
$12 = 0.002    (Arc tolerance, millimeters)
$13 = 0    (Report in inches, boolean)
$20 = 0    (Soft limits enable, boolean)
$21 = 1    (Hard limits enable, boolean)
$22 = 1    (Homing cycle enable, boolean)
$23 = 27    (Homing direction invert, mask)
$24 = 200.000    (Homing locate feed rate, mm/min)
$25 = 1000.000    (Homing search seek rate, mm/min)
$26 = 200    (Homing switch debounce delay, milliseconds)
$27 = 2.000    (Homing switch pull-off distance, millimeters)
$30 = 1000    (Maximum spindle speed, RPM)
$31 = 0    (Minimum spindle speed, RPM)
$32 = 1    (Laser-mode enable, boolean)
$100 = 200.000    (X-axis travel resolution, step/mm)
$101 = 200.000    (Y-axis travel resolution, step/mm)
$102 = 800.000    (Z-axis travel resolution, step/mm)
$103 = 200.000   
$104 = 200.000   
$110 = 1500.000    (X-axis maximum rate, mm/min)
$111 = 1500.000    (Y-axis maximum rate, mm/min)
$112 = 300.000    (Z-axis maximum rate, mm/min)
$113 = 1500.000   
$114 = 1500.000   
$120 = 200.000    (X-axis acceleration, mm/sec^2)
$121 = 200.000    (Y-axis acceleration, mm/sec^2)
$122 = 200.000    (Z-axis acceleration, mm/sec^2)
$123 = 200.000   
$124 = 200.000   
$130 = 500.000    (X-axis maximum travel, millimeters)
$131 = 500.000    (Y-axis maximum travel, millimeters)
$132 = 120.000    (Z-axis maximum travel, millimeters)
$133 = 642.000   
$134 = 642.000   
ok

**Reply by attk3** [reply]
I've already solved the problem. Thanks anyway.

**Reply by fra589** [reply]
Hi @attk3,  

Sorry for this late reply, I'm in travel and don't have normal Internet access...
What was your problem solution?

@++;
Gauthier.

**Reply by attk3** [reply]
I uncommented line 89 (#define SORT_REPORT_BY_AXIS_NAME) and now it works fine. I don't know if that's what fixed it but now there's no problem. Thanks for your reply Gauthier, have a nice day!
